talk/app/talk-cli.hoon

```diff
@@ -103,28 +206,84 @@
     ^-  (quip card _this)
     =^  cards  state
       ?-    -.sign
-        %poke-ack   [- state]:(on-agent:def wire sign)
-        %watch-ack  [- state]:(on-agent:def wire sign)
+          %poke-ack
+        ?~  p.sign  [~ state]
+        %-  (slog u.p.sign)
+        [~ state]
+      ::
+          %watch-ack
+        ?~  p.sign  [~ state]
+        =/  =tank
+          leaf/"subscribe failed from {<dap.bowl>} on wire {<wire>}"
+        %.  [~ state]
+        (slog tank u.p.sign)
       ::
           %kick
         :_  state
-        ?+  wire  ~
-          [%chat %ui ~]  ~[connect:tc]
+        ?+    wire  ~
+            [%dm ship=@ %ui ~]  
+          =/  =ship  (slav %p +<.wire)
+          (connect:tc [%ship ship])
+        ::
+            [%club id=@ %ui %writs ~]
+          =/  =club-id  (slav %uv +<.wire)
+          (connect:tc [%club club-id])
+        ::
+            [%chat ship=@ name=@ %ui %writs ~]
+          =/  =ship  (slav %p +<.wire)
+          =/  name=term  (slav %tas +>-.wire)
+          (connect:tc [%flag [ship name]])
         ==
       ::
-          %fact
-        ?+  p.cage.sign  ~|([dap.bowl %bad-sub-mark wire p.cage.sign] !!)
-            %chat-action-0
-          %-  on-chat-update:tc
-          !<(action:chat q.cage.sign)
+          %fact  
+        ?+     p.cage.sign  ~|([dap.bowl %bad-sub-mark wire p.cage.sign] !!)
+            %writ-diff
+          ?+    wire  ~
+              [%dm ship=@ %ui ~] 
```

Comment:
You should branch on wire before sign. Also, crash on unrecognized wires, those are generally implementation problems.

Reply:
Completed here: f4cbea2872403501b17b0e94f1bc66d6fe91d903